extension/chrome/elements/pgp_block_modules/pgp-block-error-module.ts

```diff
@@ -2,27 +2,24 @@
 
 'use strict';
 
-import { ApiErr } from '../../../js/common/api/shared/api-error.js';
 import { Browser } from '../../../js/common/browser/browser.js';
 import { BrowserMsg } from '../../../js/common/browser/browser-msg.js';
-import { Catch } from '../../../js/common/platform/catch.js';
-import { FormatError } from '../../../js/common/core/crypto/pgp/msg-util.js';
 import { Lang } from '../../../js/common/lang.js';
-import { PgpBlockView } from '../pgp_block.js';
+import { PgpBaseBlockView } from '../pgp_base_block_view.js';
 import { Ui } from '../../../js/common/browser/ui.js';
 import { Xss } from '../../../js/common/platform/xss.js';
 import { Str } from '../../../js/common/core/common.js';
 
 export class PgpBlockViewErrorModule {
   private debugId = Str.sloppyRandom();
 
-  public constructor(private view: PgpBlockView) {}
+  public constructor(private view: PgpBaseBlockView) {}
 
-  public renderErr = async (errBoxContent: string, renderRawMsg: string | undefined, errMsg?: string) => {
+  public renderErr = (errBoxContent: string, renderRawMsg: string | undefined, errMsg?: string) => {
     this.view.renderModule.setFrameColor('red');
     this.view.renderModule.renderErrorStatus(errMsg || 'decrypt error');
     const showRawMsgPrompt = renderRawMsg ? '<a href="#" class="action_show_raw_pgp_block">show original message</a>' : '';
-    await this.view.renderModule.renderContent(`<div class="error">${errBoxContent.replace(/\n/g, '<br>')}</div>${showRawMsgPrompt}`, true);
+    this.view.renderModule.renderContent(`<div class="error">${errBoxContent.replace(/\n/g, '<br>')}</div>${showRawMsgPrompt}`, true);
```

Comment:
This code accepts html code with possible links and `<button>` tags. The allowed origins are extension's URL and `mail.google.com` (that is the origin of the content script) -- so potentially another extension installed as content script to `mail.google.com` may send it.
Adding complex messages not containing injectable html would make the code much more complex. Should we ignore this alert? @sosnovsky @tomholub

Reply:
(this is `.renderErr` handler previously called as `errorModule.renderErr(...)` )

Reply:
So another extension can send some message to FlowCrypt, which will trigger error alert with html code possibly containing some link?

Reply:
I guess, arbitrary Javascript can also be injected too, e.g. as a handler for injected button

Reply:
Sounds like a potential security issue, is there no way to disable messages from other extensions?
Or maybe we can allow html code only for our extension URL, while `mail.google.com` can send only plain text?

Reply:
I still can't bring myself to believe that the browser extension would blindly accept messages from another extension. Have you tested this? I would expect this to have to be explicitly allowed, eg in the manifest.

If it is indeed the case, I find it very unlikely that this cannot be somehow limited when setting up the listener, or in manifest, etc.

Reply:
I found this Chrome documentation on extension messaging - https://developer.chrome.com/docs/extensions/mv3/messaging/#external

It says that for receiving messages from another extensions another method should be used - called `chrome.runtime.onMessageExternal.addListener`, so probably our current implementation with `chrome.runtime.onMessage.addListener` should get messages sent only from FlowCrypt extension?
Needs to be tested

Reply:
`chrome.runtime` messaging can't be used to send a message from content script to another part of extension.
The current codebase is using the background page as a relay.
Since the background page is scheduled to be removed (in favor of service workers) as of #1509, this way of communication won't exist in the future.
So in this PR direct communication is implemented using basic Javascript `postMessage` mechanism, which only allows to check `event.origin`. And `event.origin` of a message sent from the content scrip is `https://mail.google.com`

Reply:
Right, perhaps I was confused by a previous scenario, when messages were sent from iframes inside the content script:
```
When sending message from iframe within content script page (mail.google.com), the parent will NOT get such message directly, and it will only be delivered to background page, from where we have to relay it around.
```
We can indeed try sending messages from the content script to its inner iframes.
I'll test it shortly.